addons/monitoring/monitoring-elastic/monitoring-elastic-common/src/main/java/org/kie/kogito/monitoring/elastic/common/ElasticRegistry.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.kie.kogito.monitoring.elastic.common;
+
+import java.util.concurrent.Executors;
+
+import io.micrometer.elastic.ElasticConfig;
+import io.micrometer.elastic.ElasticMeterRegistry;
+import org.kie.kogito.monitoring.core.common.MonitoringRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ElasticRegistry {
+
+    private static final Logger logger = LoggerFactory.getLogger(ElasticRegistry.class);
+    private ElasticMeterRegistry registry;
+
+    protected ElasticRegistry() {
+    }
+
+    protected void start(ElasticConfig elasticConfig) {
+        registry = ElasticMeterRegistry.builder(elasticConfig).build();
+        MonitoringRegistry.addRegistry(registry);
+        registry.start(Executors.defaultThreadFactory());
+        logger.debug("Micrometer elastic publisher started.");
+    }
```

Comment:
I'm not sure `Executors.defaultThreadFactory()` is ideal with managed environment like Quarkus/Springboot. Can you please verify if there is a documented better option to obtain a ThreadFactory in similar environments?
In any case, please create an overloaded method to provide the instance

```suggestion
    protected void start(ElasticConfig elasticConfig) {
        start(elasticConfig, Executors.defaultThreadFactory());
    }    
    
    protected void start(ElasticConfig elasticConfig, ThreadFactory threadFactory) {
        registry = ElasticMeterRegistry.builder(elasticConfig).build();
        MonitoringRegistry.addRegistry(registry);
        registry.start(threadFactory);
        logger.debug("Micrometer elastic publisher started.");
    }
```

Reply:
Hi @danielezonca , I've refactored it: for Springboot I used https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/concurrent/ThreadPoolTaskExecutor.html, but for quarkus I did not find a (clean) way to inject a ThreadFactory. I use `Executors.defaultThreadFactory()` for it for the time being

Reply:
See https://github.com/quarkusio/quarkus/issues/13874